backend/src/openbeheer/api/views.py

```diff
@@ -261,6 +261,22 @@ class ListView[P: OBPagedQueryParams, T: Struct](MsgspecAPIView):
     expansions: Mapping[str, Expansion[T, object]] = {}
     "Maps T._expand.'attribute_name` to fetch functions"
 
+    def __init__(self, **kwargs):
+        # XXX: maybe consider making this an
+        # __init_subclass__(cls, list_type: type[T], query_type: type[P], post_type: Struct): ...
+        # so all required class vars are always set, and we can do the post method here too
+
+        super().__init__(**kwargs)
+
+        self.get = extend_schema(
+            responses={
+                "200": OBList[self.data_type],
+                "400": ZGWError,
+                "502": ExternalServiceError,
+                "504": ExternalServiceError,
+            },
```

Comment:
What happens if we then also define the schema on the child ListView?

I feel that it would be easy to forget that we shouldn't overwrite it?

Reply:
They get merged, I think. At least these operation level schemata get merged into the view level schema. I haven't tested for key collisions, but I think the last one will win, as almost everything in spectacular does in-place mutation.